torch/csrc/jit/ir.h

```diff
@@ -156,9 +157,15 @@ struct Node : public Attributes<Node> {
     return unique_;
   }
   std::string uniqueName() const {
-    if(debug_name_.size() > 0)
-      return debugName() + "_" + std::to_string(unique());
-    return std::to_string(unique());
+    std::string debugPrefix = debug_name_.size() > 0
+      ? debugName() + "_"
+      : ""
+      ;
+    std::string ret = pretty_name_ != ""
+      ? pretty_name_
+      : std::to_string(unique())
+      ;
+    return debugPrefix + ret;
```

Comment:
As seen in here, there is already a `debugName()` filed in all nodes. You can just reuse this to store the "name" taken from a Variable, and all these JIT modifications won't be necessary. If the "debug" prefix is what's worrying then we can always make it a "name"